src/main/java/redis/clients/jedis/util/Pool.java

```diff
@@ -57,126 +40,41 @@ public T getResource() {
     }
   }
 
-  protected void returnResourceObject(final T resource) {
+  public void returnResource(final T resource) {
+    if (resource == null) {
+      return;
+    }
     try {
-      internalPool.returnObject(resource);
+      super.returnObject(resource);
     } catch (Exception e) {
       throw new JedisException("Could not return the resource to the pool", e);
     }
   }
 
   public void returnBrokenResource(final T resource) {
-    if (resource != null) {
-      returnBrokenResourceObject(resource);
+    if (resource == null) {
+      return;
     }
-  }
-
-  public void returnResource(final T resource) {
-    if (resource != null) {
-      returnResourceObject(resource);
-    }
-  }
-
-  public void destroy() {
-    closeInternalPool();
-  }
-
-  protected void returnBrokenResourceObject(final T resource) {
     try {
-      internalPool.invalidateObject(resource);
+      super.invalidateObject(resource);
     } catch (Exception e) {
       throw new JedisException("Could not return the broken resource to the pool", e);
     }
   }
 
-  protected void closeInternalPool() {
+  public void destroy() {
     try {
-      internalPool.close();
+      super.close();
     } catch (Exception e) {
       throw new JedisException("Could not destroy the pool", e);
     }
   }
-  
-  /**
-   * Returns the number of instances currently borrowed from this pool.
-   *
-   * @return The number of instances currently borrowed from this pool, -1 if
-   * the pool is inactive.
-   */
-  public int getNumActive() {
```

Comment:
@sazzad16 should we keep these method to avoid backward issues?

Reply:
@gkorland all of these methods are actually available in the super class GenericObjectPool. related tests are available in src/test/java/redis/clients/jedis/tests/JedisPoolTest.java